packages/phishing-controller/src/utils.test.ts

```diff
@@ -140,3 +148,110 @@ describe('applyDiffs', () => {
     });
   });
 });
+
+describe('validateConfig', () => {
+  it('throws an error if the config is not an object', () => {
+    expect(() => validateConfig(null)).toThrow('Invalid config');
+  });
+
+  it('throws an error if the config contains a tolerance without a fuzzylist', () => {
+    expect(() => validateConfig({ tolerance: 2 })).toThrow(
+      'Fuzzylist tolerance provided without fuzzylist',
+    );
+  });
+
+  it('throws an error if the config contains an invalid name', () => {
+    expect(() => validateConfig({ name: 123 })).toThrow(
+      "Invalid config parameter: 'name'",
+    );
+  });
+
+  it('throws an error if the config contains an invalid version', () => {
+    expect(() => validateConfig({ version: { foo: 'bar' } })).toThrow(
+      "Invalid config parameter: 'version'",
+    );
+  });
+});
+
+describe('domainToParts', () => {
+  it('correctly converts a domain string to an array of parts', () => {
+    const domain = 'example.com';
+    const result = domainToParts(domain);
+    expect(result).toStrictEqual(['com', 'example']);
+  });
+
+  it('correctly converts a domain string with subdomains to an array of parts', () => {
+    const domain = 'sub.example.com';
+    const result = domainToParts(domain);
+    expect(result).toStrictEqual(['com', 'example', 'sub']);
+  });
+
+  it('throws an error if the domain string is invalid', () => {
+    // @ts-expect-error testing invalid input
+    expect(() => domainToParts(123)).toThrow('123');
+  });
+});
+
+describe('processConfigs', () => {
+  it('correctly converts a list of configs to a list of processed configs', () => {
+    const configs = [
+      {
+        allowlist: ['example.com'],
+        blocklist: ['sub.example.com'],
+        fuzzylist: ['fuzzy.example.com'],
+        tolerance: 2,
+      },
+    ];
+
+    const result = processConfigs(configs);
+
+    expect(result).toStrictEqual([
+      {
+        allowlist: [['com', 'example']],
+        blocklist: [['com', 'example', 'sub']],
+        fuzzylist: [['com', 'example', 'fuzzy']],
+        tolerance: 2,
+      },
+    ]);
+  });
+
+  it('can be called with no arguments', () => {
+    expect(processConfigs()).toStrictEqual([]);
+  });
+});
+
+describe('processDomainList', () => {
+  it('correctly converts a list of domains to an array of parts', () => {
+    const domainList = ['example.com', 'sub.example.com'];
+
+    const result = processDomainList(domainList);
+
+    expect(result).toStrictEqual([
+      ['com', 'example'],
+      ['com', 'example', 'sub'],
+    ]);
+  });
+});
+
+describe('matchPartsAgainstList', () => {
+  it('matches a domain against a list of parts', () => {
+    const domainParts = ['com', 'example'];
+    const list = [
+      ['com', 'example', 'sub'],
+      ['com', 'example'],
```

Comment:
What if you include an entry in the list that has fewer parts than `domainParts`? It should _not_ get matched, but if we are testing that it should not match a domain with more parts, it seems we should also test that it shouldn't match one with fewer.

``` suggestion
      ['com'],
      ['com', 'example'],

Reply:
Hmm okay. It sounds like we may be testing this advertently. If that is the case, then I'm okay with making the tests more robust in another PR.